src/plot_api/plot_api.js

```diff
@@ -105,7 +105,8 @@ exports.plot = function(gd, data, layout, config) {
 
     // hook class for plots main container (in case of plotly.js
     // this won't be #embedded-graph or .js-tab-contents)
-    d3.select(gd).classed('js-plotly-plot', true);
+    d3.select(gd).classed('js-plotly-plot', true)
+      .style('position', 'relative');
```

Comment:
Hmm. I'll call that a breaking change.

We could turn this on just when `responsive` is set to `true`. But then again, are there any other solutions?

Reply:
Yes, that's certainly a breaking change and it breaks a lot of tests anyway. 

I have another branch in which I fix the responsive issues for both WebGL #3485 and the modebar https://github.com/plotly/plotly.js/issues/3499 that doesn't rely on modifying the parent. I will submit a new separate PR with a cleaner commit history and I will close the current one.